src/Microsoft.OData.Client/ALinq/ApplyQueryOptionExpression.cs

```diff
@@ -39,6 +45,47 @@ public override ExpressionType NodeType
 		/// </summary>
 		internal List<Aggregation> Aggregations { get; private set; }
 
+		/// <summary>
+		/// Adds the conjuncts to the filter expressions
+		/// </summary>
+		internal void AddPredicateConjuncts(IEnumerable<Expression> predicates)
+		{
+			this.filterExpressions.AddRange(predicates);
+		}
+
+		internal ReadOnlyCollection<Expression> PredicateConjuncts
+		{
+			get
+			{
+				return new ReadOnlyCollection<Expression>(this.filterExpressions);
+			}
+		}

```

Comment:
Is this property access multiple times? And if so, is it expected to return a new collection each time?

Reply:
@habbes ReadOnlyCollection is just a wrapper on top of the `filterExpressions` list. It does not create a new list
https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlycollection-1?view=net-5.0#examples